src/containers/App.tsx

```diff
@@ -57,33 +60,52 @@ const App = (props: IProps) => {
     );
   };
 
-  return (
-    <View style={styles.container}>
-      <Button
-        onPress={props.addCounter}
-        title={'Add Counter'}
-        color={styles.button.color}
-        accessibilityLabel={'Add Counter'}
-      />
-      <View style={styles.separator} />
-      <FlatList
-        data={props.counters}
-        renderItem={renderCounter}
-        keyExtractor={getKey}
-      />
-    </View>
-  );
+  if (props.loadingStatus === LoadingStatus.Success) {
+    return (
+      <View style={styles.container}>
+        <Button
+          onPress={props.addCounter}
+          title={'Add Counter'}
+          color={styles.button.color}
+          accessibilityLabel={'Add Counter'}
+        />
+        <View style={styles.separator} />
+        <FlatList
+          data={props.counters}
+          renderItem={renderCounter}
+          keyExtractor={getKey}
+        />
+      </View>
+    );
+  } else {
+    let text;
```

Comment:
This loading state should be a bit prettier on the UI side. Right now it is in the top left of the screen, which is quite difficult to see. I would want it in the middle and somewhat stylized.